### PR TITLE
Dt 1439 prison location multiple custodial events

### PR DESCRIPTION
**Pull request by andymarke**
Previous if an offender had multiple sentences (events) we would not update the prison location on any of the events but return an error, this change will updated each active custodial event (based on current rules around custodial status)

### CODE REVIEW

File: src/test/java/uk/gov/justice/digital/delius/transformers/ConvictionTransformerTest.java [hunk 1]
```diff
@@ -436,13 +435,6 @@ private Offence anOffence() {
                 .build();
     }
 
-    private uk.gov.justice.digital.delius.data.api.Offence anApiMainOffence() {
```
Comment (andymarke) [inline src/test/java/uk/gov/justice/digital/delius/transformers/ConvictionTransformerTest.java:439]: no used

File: src/test/java/uk/gov/justice/digital/delius/transformers/ConvictionTransformerTest.java [hunk 2]
```diff
@@ -464,16 +456,8 @@ private Disposal aDisposal() {
                 .softDeleted(0L);
     }
 
-    private CourtCase aCourtCase() {
```
Comment (andymarke) [inline src/test/java/uk/gov/justice/digital/delius/transformers/ConvictionTransformerTest.java:467]: not used

File: src/main/java/uk/gov/justice/digital/delius/data/api/Custody.java [hunk 3]
```diff
@@ -20,4 +22,6 @@
     private CustodyRelatedKeyDates keyDates;
     @ApiModelProperty(value = "Custodial status")
     private KeyValue status;
+    @ApiModelProperty(value = "Date when related sentence started")
```
Comment (andymarke) [inline src/main/java/uk/gov/justice/digital/delius/data/api/Custody.java:25]: added so custody objects can be distinguished when two sentences (events) combine to single time in prison

File: src/test/java/uk/gov/justice/digital/delius/service/CustodyServiceTest.java [hunk 4]
```diff
@@ -413,17 +477,6 @@ public void willCreateTelemetryEventWhenConvictionNotFound() {
         }
 
 
-        @Test
-        @DisplayName("will add multiple events telemetry when more than one active custodial events found")
```
Comment (andymarke) [inline src/test/java/uk/gov/justice/digital/delius/service/CustodyServiceTest.java:417]: moved (and amended) to nested class below

File: src/main/java/uk/gov/justice/digital/delius/service/CustodyService.java [hunk 5]
```diff
@@ -92,18 +96,21 @@ public Custody updateCustodyPrisonLocation(final String nomsNumber,
...
+                    telemetryClient.trackEvent("P2PTransferPrisonUpdated", add(telemetryProperties, "updatedCount", String.valueOf(success.custodyRecordsUpdated.size())), null);
                     break;
                 case NoUpdateRequired:
                     telemetryClient.trackEvent("P2PTransferPrisonUpdateIgnored", telemetryProperties, null);
                     break;
             }
-            return success.custody;
+            return success.custodyRecordsUpdated
+                .stream()
+                .max(Comparator.comparing(Custody::getSentenceStartDate))
```
Comment (mikehalmamoj) [inline src/main/java/uk/gov/justice/digital/delius/service/CustodyService.java:112]: Just wondering why max?

**Review by mikehalmamoj — APPROVED** [review]

File: src/main/java/uk/gov/justice/digital/delius/service/CustodyService.java (hunk 5 above)
Comment (andymarke) [reply src/main/java/uk/gov/justice/digital/delius/service/CustodyService.java:112]: We only return one of the custody records, so max essentially is the latest one which is most appropriate.

**Pull request merged**